functions/backup.bash

```diff
@@ -237,9 +215,156 @@ amanda_setup() {
 
       create_backup_config "${config}" "${backupuser}" "${adminmail}" "${tapes}" "${size}" AWS "${S3site}" "${S3bucket}" "${S3accesskey}" "${S3secretkey}"
     fi
+
+    whiptail --title "Operation Successful!" --msgbox "$successtext" 15 80
+  fi
+
+  echo "OK"
+}
+
+
+## "raw" copy partition using dd or mount and use rsync to sync "diff"
+## Valid arguments: "raw", "diff"
+## Periodically activated by systemd timer units (raw copy on 1st of month, rsync else)
+##
+##    mirror_SD(String method)
+##
+mirror_SD() {
+  local src
+  local dest
+  local start
+  local syncMount="/storage/syncmount"
+
+
+  src=/dev/mmcblk0
+  if [[ -n "$INTERACTIVE" ]]; then
+    select_blkdev "^sd" "Setup SD mirroring" "Select the USB attached disk device to copy the internal SD card data to"
+    # shellcheck disable=SC2154
+    if [[ -z "$retval" ]]; then return 0; fi
+    dest="/dev/$retval"
+  else
+    dest="${backupdrive:-/dev/sda}"
+  fi
+  if [[ "$1" == "raw" ]]; then
+    echo "Creating a raw partition copy, be prepared this may take long such as 20-30 minutes for a 16 GB SD card"
+    if ! cond_redirect dd if="${src}" bs=1M of="${dest}"; then echo "FAILED (raw device copy)"; return 1; fi
+    echo "OK"
+    return 0;
   fi
 
+  if [[ "$1" == "diff" ]]; then
+    if [[ ! -d $syncMount ]]; then
+      mkdir -p "$syncMount"
+    fi
+    if [[ -n "$INTERACTIVE" ]]; then
+      select_blkdev "^-sd" "select partition" "Select the partition to copy the internal SD card data to"
+      dest="/dev/$retval"
+    else
+      dest=${dest}2
+    fi
+    # Auswahl des src dir nötig ?? nein immer "/"
+    # Auswahl des dest dir nötig ?? nein immer syncmount und dest wird oben ausgewählt
+    #if [[ -n "$INTERACTIVE" ]]; then
+    mount "$dest" "${syncMount:-/storage/syncmount}"
+    rsync --one-file-system -avRh "/" "$syncMount"
+    if ! umount "$syncMount" &> /dev/null; then
```

Comment:
Use `()` in if statements if output redirection is used to help avoid ambiguity.

Reply:
what's ambiguous about that ?
() is subshell. Not sure it would work the same.

Reply:
it does.

Reply:
ok then